### PR TITLE
feat: add terramate.config.run.check_gen_code config

### PR DESCRIPTION
**Pull request by katcipis**
Add new configs to run block:

```
check_gen_code = true
```

When the attribute is not defined in the run block it defaults to true.

### CODE REVIEW

File: hcl/hcl.go [hunk 1]
```diff
@@ -1062,8 +1068,30 @@ func parseRunConfig(runCfg *RunConfig, runBlock *ast.MergedBlock) error {
...
+				"failed to evaluate terramate.config.run.%s attribute", attr.Name,
+			))
+
+			continue
+		}
+
+		switch attr.Name {
+		case "check_gen_code":
+			if value.Type() != cty.Bool {
+				errs.Append(errors.E(attr.Expr.Range(),
```
Comment (katcipis) [inline hcl/hcl.go:1083]: Can use the new error creation function after we merge #458

File: test/hcl.go [hunk 2]
```diff
@@ -106,6 +106,12 @@ func assertTerramateRunBlock(t *testing.T, got, want *hcl.RunConfig) {
 		return
 	}
 
+	if want.CheckGenCode != got.CheckGenCode {
```
Comment (i4ki) [inline test/hcl.go:109]: now we can do:

```
assert.IsTrue(t, want.CheckGenCode == got.CheckGenCode, 
    "want.Run.CheckGenCode %v != got.Run.CheckGenCode %v", want.CheckGenCode, got.CheckGenCode)
```
but just an idea... feel free to ignore.

**Review by i4ki — APPROVED** [review]
lgtm

File: test/hcl.go (hunk 2 above)
Comment (katcipis) [reply test/hcl.go:109]: Forgot we now have this, wiill do, thanks :heart:

Comment (katcipis) [reply test/hcl.go:109]: I even looked for it but I used go doc and I think go doc showed me the version on the go.mod, which doesn't have this, so I thought I was just going crazy :laughing:

**Review by i4ki — APPROVED** [review]

File: hcl/hcl.go (hunk 1 above)
Comment (katcipis) [reply hcl/hcl.go:1101]: Yeah I just fix it then, already working on it here :+1:

**Pull request merged**